### PR TITLE
perlapi: Combine all av_len synonyms into one entry

### PR DESCRIPTION
**Pull request by khwilliamson**

### CODE REVIEW

**Comment by richardleach** [comment]
LGTM. 

Only query is will it look a bit odd now to document the return type for just `AvFILL`?

**Comment by tonycoz** [comment]
> Only query is will it look a bit odd now to document the return type for just `AvFILL`?

The others have entries in `embed.fnc` so `autodoc.pl` picks up the return and parameter types from there.

**Review by tonycoz — DISMISSED** [review]

**Comment by khwilliamson** [comment]
I realized that it would be better if the issue about the signatures didn't come up at all.  So, I changed autodoc to allow a signature declaration line next to the macro definition which will be pulled in by the actual autodoc line.

File: autodoc.pl [hunk 1]
```diff
@@ -1797,12 +1799,21 @@ sub add_defn  {
     next if $file =~ m! ^ ( cpan | dist | ext ) / !x
          && ! defined $extra_input_pods{$file};
 
-    open F, '<', $file or die "Cannot open $file for docs: $!\n";
-    autodoc(\*F,$file);
-    close F or die "Error closing $file: $!\n";
+    if (0 && $file =~ /\.h/) {
```
Comment (tonycoz) [inline autodoc.pl]: Condition always false, presumably left over from debugging.

Comment (khwilliamson) [reply autodoc.pl]: yes; now fixed

**Review by tonycoz — APPROVED** [review]
This looks fine.

I wondered if there should be an update to documentation for the autodoc pod syntax, but I didn't find anything to update.

**Comment by khwilliamson** [comment]
The new syntax is documented in `embed.fnc`

**Pull request merged**